pydantic/env_settings.py

```diff
@@ -38,6 +46,12 @@ def _build_environ(self) -> Dict[str, Optional[str]]:
         else:
             env_vars = {k.lower(): v for k, v in os.environ.items()}
 
+        env_file = _env_file if _env_file != env_file_sentinel else self.__config__.env_file
+        if env_file is not None:
+            env_path = Path(env_file)
+            if env_path.is_file():
+                env_vars = {**env_vars, **read_env_file(env_path, self.__config__.case_sensitive)}
```

Comment:
```suggestion
                env_vars = {**read_env_file(env_path, self.__config__.case_sensitive), **env_vars}
```
my mistake, environment variables should definitely take precedence over env files.